R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+#' @param destination_folder Folder path where the downloaded files need to be stored. By default, it uses the R working directory to store files.
+#' @param index The name of the time attribute that should be used as the index when creating the tsibble. If doesn't provide, it tries to find a valid index within the data. If there is no valid index, then a data frame will be returned instead of a tsibble.
+#' @param key The name of the attribute that should be used as the key when creating the tsibble. The default value is "series_name" which is the key of the datasets in our repository. If doesn't provide, a data frame will be returned instead of a tsibble.
+#' @param value_column_name Any name that is preferred to have as the name of the column containing series values in the returning tsibble.
+#'
+#' @return This function returns a tsibble or a data.frame based on the validity of given key and index values.
+#' @importFrom utils download.file unzip
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
+download_data_zenodo <- function(record_id, file_name, destination_folder = '', index = NULL, key = "series_name", value_column_name = "series_value"){
+  
+  if(is.null(record_id) | is.na(as.numeric(record_id)))
+    stop("Please provide the ID of the dataset that you need to download. This should be a numerical value.")
+  
+  if(is.null(file_name) | is.na(as.character(file_name)))
+    stop("Please provide the file name of the dataset that you need to download. This should be a character/string value.")
+  
+  # Check whether the destination folder exists. If not create a folder at the specified path
+  if(destination_folder != '' & !file.exists(destination_folder)){
+    tryCatch({
+      dir.create(destination_folder)
+    }, error = function(e) {   
+      message(e)
+      stop("Please provide a valid path/name for destination folder")
+    })
+  }
+    
+  if(destination_folder == ''){  
+    dest_file <- paste0(file_name, ".zip")
+    tsf_file <- paste0(file_name, ".tsf")
+  }else{  
+    dest_file <- file.path(destination_folder, paste0(file_name, ".zip"))
+    tsf_file <- file.path(destination_folder, paste0(file_name, ".tsf"))
+  }
+  
+  
+  if(!file.exists(tsf_file)){ # Check whether the required .tsf file is already there at the specified destination folder
+    if(!file.exists(dest_file)){ # Check whether the required .zip file is already there at the specified destination folder. If not download the .zip file from Zenodo
+      tryCatch({
+        download.file(paste0("https://zenodo.org/record/", record_id, "/files/", file_name, ".zip"), destfile = dest_file)
```

Comment:
Functions with side effects (such as downloading data) need to be very explicit about where and how the data is stored.
Is this function hoping to store the data locally for the next time it is obtained? If so, `{rappdirs}` (https://rappdirs.r-lib.org/) should be used to conform with CRAN policies on data storage.

Reply:
**rappdirs::user_data_dir()** was used to store the downloaded files.